ekncontent/ekncontent/eknc-domain.c

```diff
@@ -830,8 +858,13 @@ on_xapian_query_response (GObject *source,
           return;
         }
 
+      /* Note that we pass another closure here instead of just the task
```

Comment:
I'm *sure* there must be a better way to do this, and I'd even bet there is an example of it somewhere in GNOME code. In particular I don't like the separate success/error callbacks, as that duplicates `g_task_return_pointer()`/`g_task_return_error()`.

It seems to me the only data you need in `on_object_response()` is the `EkncDomain` (which would be the task's "source object") and the models array plus an index into it.